File: client-java/controller-api/src/main/java/org/evomaster/client/java/controller/api/dto/database/execution/epa/Enabled.java
package org.evomaster.client.java.controller.api.dto.database.execution.epa;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Enabled {

    @JsonProperty("associatedRestAction")
    public RestAction associatedRestAction;

    @JsonProperty("enabledRestActions")
    public RestActions enabledRestActions;

    public Enabled() {
        this.associatedRestAction = null;
        this.enabledRestActions = new RestActions();
    }

    public Enabled(@JsonProperty("associatedRestAction") RestAction lastRestAction,
                   @JsonProperty("enabledRestActions") RestActions enabledRestActions) {
        this.associatedRestAction = lastRestAction;
        this.enabledRestActions = enabledRestActions;
    }
}

File: core/src/main/kotlin/org/evomaster/core/problem/rest/epa/Enabled.kt
package org.evomaster.core.problem.rest.epa

import org.evomaster.client.java.controller.api.dto.database.execution.epa.RestAction
import org.evomaster.client.java.controller.api.dto.database.execution.epa.RestActions


data class Enabled(
    var associatedRestAction: RestAction,
    var enabledRestActions: RestActions = RestActions()
)